syft/src/main/java/org/openmined/syft/execution/SyftJob.kt

```diff
@@ -69,21 +68,22 @@ class SyftJob internal constructor(
                 modelName,
                 version,
                 worker,
-                config,
-                JobRepository(
-                    JobLocalDataSource(),
-                    JobRemoteDataSource(config.getDownloader())
-                )
+                config
             )
         }
     }
 
-    val jobId = JobID(modelName, version)
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    internal val jobRepository = JobRepository.create(config, modelName, version)
+
     internal var cycleStatus = AtomicReference(CycleStatus.APPLY)
+
     internal val requiresSpeedTest = AtomicBoolean(true)
+
     private val isDisposed = AtomicBoolean(false)
 
     private val plans = ConcurrentHashMap<String, Plan>()
+
```

Comment:
Send all of these variables into jobModel. It should store plans, protocols, dispose state and requires speed test along with model name, model version and jobId